models.py

```
@@ -26,22 +28,129 @@ def get_attr_by_path(self, path: str, sep: str = "__"):
 
         return current_level
 
-
     def get_indicators(self) -> dict:
         indicators = {}
         for indicator in self.indicators:
             value = self.get_attr_by_path(indicator["id"])
             cmp = indicator["not"] if isinstance(indicator["not"], list) else [indicator["not"]]
             indicators[f"has_{indicator['id']}"] = all(value != c for c in cmp)
+
         return indicators
 
+    def get_prefix_or_fallback_from(self, url: str, message: str) -> str:
+        m = re.match("^(.*/)[^/]+$", url)
+
+        if m:
+            return m.group(1)
+
+        return message
+
+    def compute_prefix_harvest_remote_id(self) -> str:
+        missing_prefix_message = 'Préfixe manquant'
+
+        try:
+            if (self.payload is None):
+                raise KeyError
+
+            harvest = self.payload['harvest']
+
+            if (harvest is None):
+                raise KeyError
+
+            remote_id = harvest['remote_id']
+
+            if (remote_id is None):
+                raise KeyError
+        except KeyError:
```

J'ai du mal a saisir l'interêt ? Pourquoi ne pas directement `return missing_prefix_message` au lieu de lever une exception rattrapée aussitôt ?

---

Pour minimiser les `return`

---

C'est plutôt un bon pattern IMO de return asap (~ fail fast), non ? De même que de n'utiliser les exceptions que pour les cas exceptionnels (= inattendus ou anormaux).
Mais c'est du nitpick, la logique reste correcte dans les deux cas.